protocol/daemons/liquidation/client/client.go

```diff
@@ -65,70 +85,152 @@ func Start(
 	liquidationServiceClient := api.NewLiquidationServiceClient(daemonConn)
 
 	ticker := time.NewTicker(time.Duration(flags.Liquidation.LoopDelayMs) * time.Millisecond)
-	for ; true; <-ticker.C {
-		if err := RunLiquidationDaemonTaskLoop(
-			ctx,
-			flags.Liquidation,
-			subaccountQueryClient,
-			clobQueryClient,
-			liquidationServiceClient,
-		); err != nil {
-			// TODO(DEC-947): Move daemon shutdown to application.
-			logger.Error("Liquidations daemon returned error", "error", err)
-		}
-	}
+	done := make(chan bool)
+
+	s := &SubTaskRunnerImpl{}
+	StartLiquidationsDaemonTaskLoop(
+		ctx,
+		c,
+		s,
+		flags,
+		ticker,
+		done,
+		subaccountQueryClient,
+		clobQueryClient,
+		liquidationServiceClient,
+		logger,
+	)
 
 	return nil
 }
 
-// RunLiquidationDaemonTaskLoop contains the logic to communicate with various gRPC services
-// to find the liquidatable subaccount ids.
-func RunLiquidationDaemonTaskLoop(
+// StartLiquidationsDaemonTaskLoop contains the logic to periodically run the liquidations daemon task.
+func StartLiquidationsDaemonTaskLoop(
 	ctx context.Context,
-	liqFlags flags.LiquidationFlags,
+	client *Client,
+	s SubTaskRunner,
+	flags flags.DaemonFlags,
+	ticker *time.Ticker,
+	done <-chan bool,
 	subaccountQueryClient satypes.QueryClient,
 	clobQueryClient clobtypes.QueryClient,
 	liquidationServiceClient api.LiquidationServiceClient,
-) error {
+	logger log.Logger,
+) {
```

Comment:
Added the `StartLiquidationsDaemonTaskLoop` as an entrypoint for testing that encapsulates the looping logic formerly in the `Start` method. This method takes a ticker as well as a newly introduced `done` channel that signals loop termination. 

This is similar to the design we use in the[ pricefeed daemon,](https://github.com/dydxprotocol/v4-chain/blob/755b0b928be793072d19eb3a1608e7a2503f396a/protocol/daemons/pricefeed/client/client.go#L205) which also supports terminating the daemon on command.